<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Storing Data (Need) Exercises</span></div>

In [9]:
import numpy as np

# Exercise 1

Two similarities between lists and arrays.
1. Both are used to store information to be used later on
2. *Order matters*
   
Two differences between lists and arrays.
1. Arrays work on individual elements while lists work on the entire list of information
2. There are more functions available for arrays than lists


What is a dictionary?
1. Storing of data based on keys and values

# Exercise 2

In [1]:
py_list = ["a1", "b2", "c3", "d4", "e5", "f6", "g7", "h8", "i9", "j10"]
print(py_list[0])                    # Prints 'a1'
print(py_list[2])                    # Prints 'c3'
print(py_list[4])                    # Prints 'e5'
print(py_list[6])                    # Prints 'g7'
print(py_list[8])   # Prints 'i9'

a1
c3
e5
g7
i9


# Exercise 3

In [3]:
elements = ['Hydrogen',
            'Helium', 'Lithium',
            'Beryllium', 'Boron', 'Carbon',
            'Nitrogen', 'Oxygen',
            'Fluorine',
            'Neon']

In [5]:
print(elements[4])

Boron


In [6]:
print(elements[-4])

Nitrogen


# Exercise 4

In [7]:
numbers=[45, 60, 1, 30, 96, 1, 96, 57, 16, 1,
        99, 62, 86, 43, 42, 60, 59, 1, 1, 35,
        83, 47, 34, 28, 68, 23, 22, 92, 1, 79,
        1, 29, 94, 72, 46, 47, 1, 74, 32, 20,
        8, 37, 35, 1, 89, 29, 86, 19, 43, 61]                  

In [13]:
np_arrays_numbers = np.array(numbers)

In [16]:
question = np_arrays_numbers == 1

In [19]:
question.sum()

np.int64(9)

# Exercise 5

In [20]:
x_1 = [1, 4, 9, 16, 25, 36, 49, 64, 81, 100]
x_2 = [1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]

In [22]:
array_x_1 = np.array(x_1)
array_x_2 = np.array(x_2)

In [68]:
mean_x_1 = array_x_1.mean()

In [69]:
print(mean_x_1)

38.5


In [70]:
mean_x_2 = array_x_2.mean()

In [71]:
print(mean_x_2)

302.5


In [28]:
array_x_1.std()

np.float64(32.41990129534635)

In [29]:
array_x_2.std()

np.float64(326.0893282522444)

$\gamma$ = $\dfrac{\sum_{i=1}^{n} (x_i-\mu_X)(y_i-\mu_Y)}{\sqrt{\sum_{i=1}^{n} (x_i-\mu_X)^2 \sum_{i=1}^{n} (y_i-\mu_Y)^2}}$

In [67]:
x1_mean_diff = array_x_1-mean

In [65]:
x2_mean_diff = array_x_2-array_x_2.mean()

In [53]:
top = x1_mean_diff*x2_mean_diff

In [50]:
numerator = sum(top)

In [51]:
x1_mean_diff_sqred = x1_mean_diff**2

In [52]:
x2_mean_diff_sqred = x2_mean_diff**2

In [54]:
sum_x1_bottom = sum(x1_mean_diff_sqred)

In [55]:
sum_x2_bottom = sum(x2_mean_diff_sqred)

In [56]:
bottom = sum_x1_bottom*sum_x2_bottom

In [58]:
denominator = np.sqrt(bottom)

In [60]:
coefficient = numerator/denominator

In [61]:
print(coefficient)

-1.1266074094434364


In [62]:
np.corrcoef(array_x_1,array_x_2)

array([[1.        , 0.98717966],
       [0.98717966, 1.        ]])

1 is not significant as the matrix compares the arrays with itself and with the other array 
hence the correlation coefficient is 0.98717966